src/components/Navigator/QuickNavigationModal.vue

```diff
@@ -9,180 +9,387 @@
-    fuzzyMatch: ({ debouncedInput, symbols, processedInputRegex }) => (
-      symbols.map((symbol) => {
-        const match = processedInputRegex.exec(symbol.title.toLowerCase());
+      // Remove space-character
+      this.debouncedInput = value.replace(/\s/g, '');
+    }, 250),
+    endingPointHook() {
+      // Reset selected symbol to the first one of the list
+      this.focusedIndex = 0;
+    },
+    fuzzyMatch({ debouncedInput, symbols, processedInputRegex }) {
+      return symbols.map((symbol) => {
+        const match = processedInputRegex.exec(symbol.title);
         // Dismiss if symbol isn't matched
         if (!match) return false;
 
         const matchLength = match[0].length;
         const inputLength = debouncedInput.length;
         // Dismiss if match length is greater than 3x the input's length
         if (matchLength > inputLength * 3) return false;
-
         return ({
           title: symbol.title,
           path: symbol.path,
+          parents: this.getParents(symbol.parent),
+          type: symbol.type,
           inputLengthDifference: symbol.title.length - inputLength,
           matchLength,
           matchLengthDifference: matchLength - inputLength,
           start: match.index,
           substring: match[0],
         });
-      }).filter(Boolean)
-    ),
+      }).filter(Boolean);
+    },
+    handleKeyEnter(event) {
+      if (
+        (event.srcElement.id !== 'filter-input' && event.srcElement.id !== MatchId)
+        || this.noResultsWereFound
+      ) return;
+      this.$router.push(this.filteredSymbols[this.focusedIndex].path);
+      this.closeQuickNavigationModal();
+    },
     orderSymbolsByPriority(matchingSymbols) {
       return matchingSymbols.sort((a, b) => {
         // Shortests symbol match title have preference over larger titles
         if (a.matchLengthDifference > b.matchLengthDifference) return 1;
         if (a.matchLengthDifference < b.matchLengthDifference) return -1;
-        // Shortests symbol title have preference over larger titles
-        if (a.inputLengthDifference > b.inputLengthDifference) return 1;
-        if (a.inputLengthDifference < b.inputLengthDifference) return -1;
         // Matches at the beginning of string have relevance over matches at the end
         if (a.start > b.start) return 1;
         if (a.start < b.start) return -1;
+        // Shortests symbol title have preference over larger titles
+        if (a.inputLengthDifference > b.inputLengthDifference) return 1;
+        if (a.inputLengthDifference < b.inputLengthDifference) return -1;
         return 0;
       });
     },
-  },
-  watch: {
-    userInput: 'debounceInput',
+    scrollIntoView() {
+      this.$refs.match[this.focusedIndex].scrollIntoView({
+        block: 'nearest',
+        inline: 'start',
+      });
+    },
+    startingPointHook() {
+      this.focusedIndex = this.totalItemsToNavigate - 1;
+    },
   },
 };
 </script>
 
-<style scoped lang="scss">
+<style lang="scss">
```

Comment:
Is there a specific reason why you removed the `scoped` attribute for these styles? It can be a little dangerous since these styles might now interfere with other components on the page that might contain the same classnames (even though it does seem like the majority here are very specific).

Reply:
I removed it to style the inner `Filter Input` component, but it doesn't seem necessary anymore with the latest changes, thanks! Added in https://github.com/apple/swift-docc-render/pull/396/commits/62db4a637515e8fd67f83266a028ced48f825d01